src/main/java/it/gov/pagopa/paymentupdater/repository/PaymentRepository.java

```diff
@@ -15,5 +16,8 @@ public interface PaymentRepository extends MongoRepository<Payment, String> {
 	Optional<Payment> getPaymentByNoticeNumberAndFiscalCode(String noticeNumber, String fiscalCode);
 
 	@Query("{'rptId':?0}")
-	Payment getPaymentByRptId(String rptId);
+	List<Payment> getPaymentByRptId(String rptId);
+
+	int countFindById(String id);
```

Comment:
countById?

Reply:
As mentioned, if a duplicate message arrives, we check the existence of the payment. Could this check affect performance?

Reply:
Nope. I've suggested a method rename because it can be misunderstood. `CountFind..` seems that multiple find operations are processed here.

Reply:
fixed